sdk/core/System.ClientModel/samples/ModelReaderWriter.md

```diff
@@ -53,4 +53,35 @@ string json = @"{
       ""z"": 3
     }";
 OutputModel? model = JsonSerializer.Deserialize<OutputModel>(json, options);
+```
+
+## Using a Proxy
+
+In more advanced scenarios a library user might want to override the behavior of how a model is read or written.
+In this case you can implement your own class which implements the same `IPersistableModel<T>` or `IJsonModel<T>` and register it with the `ModelReaderWriterOptions`.
```

Comment:
> the same `IPersistableModel<T>` or `IJsonModel<T>`

this is a little unclear -- would it be possible ground this in an example?  e.g. something like -- "for example, to add a proxy for a model `InputModel` that implements `IJsonModel<InputModel>`, you would add a type such as `InputModelProxy` that also implements `IJsonModel<InputModel>`.  This clarifies a bit what "the same `IJsonModel<T>` means.